Stack item quantities of items looted with the `LootNPCsPopup`

In3luki opened the pull request:
This has the same logic as `PhysicalItemPF2e#isStackableWith`.

![Recording 2023-09-15 at 13 53 27](https://github.com/foundryvtt/pf2e/assets/41452412/6648b57e-d730-4083-bb81-112a44f158dc)

I've also fixed the popup UI after recording the video:

![image](https://github.com/foundryvtt/pf2e/assets/41452412/e7b8be99-0f7a-4b94-805e-047cdeb586fe)

Closes #8796

stwlam commented:
Can this reuse `ActorPF2e#findStackableItem`?

In3luki commented:
hmm, it might be possible to build a list of stackable items first and then only turn one of them to source and add the quantity of the rest.
I'll think about it,

In3luki commented:
Updated to use existing methods to determine stackability. Also now updates existing items on the loot actor if a stackable item is looted.

CarlosFdez commented on src/module/actor/sheet/loot/loot-npcs-popup.ts:
```diff
@@ -34,14 +36,51 @@ export class LootNPCsPopup extends FormApplication<ActorPF2e> {
-                itemData.push(...currentSourceItemData);
-                const idsToDelete = currentSourceItemData.map((item) => item._id);
+                for (const item of currentSource.inventory) {
+                    const stackableItem = currentSource.findStackableItem(this.object, item);
```
Should probably be this.object.findStackableItem() to keep it consistent with other uses of that same function.

After this I'm probably going to move findStackableItem() to actor.inventory and remove the first parameter, and add a deprecation warning for the existing function.

In3luki replied:
yeah, I was wondering why wasn't at least a static method. I'll change it to this.object for now.

CarlosFdez commented on src/module/actor/sheet/loot/loot-npcs-popup.ts:
```diff
@@ -34,14 +36,51 @@ export class LootNPCsPopup extends FormApplication<ActorPF2e> {
+        // Create new items in the loot actor
+        if (newItems.length > 0) {
+            const stacked: PhysicalItemPF2e[] = [];
+            for (const item of newItems) {
```
This entire for looks like it could be a reduce. With a comment saying to merge stackable items within the same incoming batch.

In3luki replied:
will do

In3luki commented:
Updated with applied suggestions. I've also moved the deletion of the source items to the end of the method to make it easier to recover from one of the other steps failing for whatever reason.
